### PR TITLE
[nl] accept specific EN words in NL

### PR DESCRIPTION
**Pull request by mark-baas**

### CODE REVIEW

File: languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/MorfologikDutchSpellerRule.java [hunk 1]
```diff
@@ -72,4 +81,46 @@ protected String getProhibitFileName() {
...
+        String message = "Het lijkt erop dat dit een foutief gespeld Engels woord is.";
+        String shortMessage = "Mogelijke fout in Engels woord";
+        List<String> suggestions = englishSpeller.getSuggestions(word);
+        RuleMatch ruleMatch = new RuleMatch(this, sentence, startPos, startPos + word.length(),
+          message, shortMessage, suggestions);
+        return Collections.singletonList(ruleMatch);
+      } else {
+        return Collections.emptyList();
+      }
+      */
```
Comment (jaumeortola) [inline languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/MorfologikDutchSpellerRule.java:121]: I thought that `englishSpeller.isMisspelled(word)` would be used to check the words around `_FOREIGN_ENGLISH` or with this POS tag, but it is not. 
With this configuration, any nonsense word around a word in `ENGLISH_WORDS` will be ignored. Is that what we want?

Comment (jaumeortola) [reply languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/MorfologikDutchSpellerRule.java:121]: I expected something like: 

```
if (tokens[idx].hasPosTag("_FOREIGN_ENGLISH") && !englishSpeller.isMisspelled(tokens[idx].getToken()) {
  return Collections.emptyList();
}
```

**Review by jaumeortola — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/MorfologikDutchSpellerRule.java [hunk 2]
```diff
@@ -72,4 +81,46 @@ protected String getProhibitFileName() {
...
+        plainTextDicts.add(fileName);
+      }
+    }
+    if (getLanguageVariantSpellingFileName() != null && getDataBroker().resourceExists(getLanguageVariantSpellingFileName())) {
+      languageVariantPlainTextDict = getLanguageVariantSpellingFileName();
+    }
+    MorfologikMultiSpeller englishSpeller = new MorfologikMultiSpeller(englishDictFilepath, plainTextDicts, languageVariantPlainTextDict,
+            userConfig, 1, language);
+    setConvertsCase(englishSpeller.convertsCase());
+  }
```
[the remark this answers is not in the document]
Comment (jaumeortola) [reply languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/MorfologikDutchSpellerRule.java:101]: Anyway, that is not critical. It is optional.

File: languagetool-language-modules/nl/src/main/resources/org/languagetool/resource/nl/disambiguation.xml [hunk 3]
```diff
@@ -10,6 +10,8 @@ Copyright (C) 2008-2024 Ruud Baars
 <!ENTITY abbrevMonths "jan|feb|(maa|mar|mrt)|apr|jun|jul|aug|sep|oct|nov|dec">
 <!ENTITY Maand '<token regexp="yes">&months;|&abbrevMonths;</token>'>
 <!ENTITY WeekDag '<token regexp="yes">&weekdays;</token>'>
+<!ENTITY ENGLISH_WORDS "up|at|mother|father|nothing|press|dead|selected|left|width|height|within|love|life|loser?s?|secret|influential|most|king|week|function|length|yeah|welcome|these|but|top|too|return|go|goes|baby|shop|appeal|trade|toys?|or|who|years?|from|about|there|also|be|boom|why|this|drink|flash|revival|not|what|with|wasn?|hasn?|hadn?|few|must|through|try|off|out|cut|sessions?|spot|games?|are|by|couldn?|can|wouldn?|might|the|and|for|your|back|that|you|he|she|its?|they|him|her|them|good|bad">
+<!ENTITY TEST_ENGLISH_WORDS "indubitably">
```
Comment (jaumeortola) [inline languagetool-language-modules/nl/src/main/resources/org/languagetool/resource/nl/disambiguation.xml:15]: `ENGLISH_WORDS` is not being used anywhere, right? I assume you are doing a test with `TEST_ENGLISH_WORDS` only, and later you will use `ENGLISH_WORDS`.

Comment (mark-baas) [reply languagetool-language-modules/nl/src/main/resources/org/languagetool/resource/nl/disambiguation.xml:15]: That's right.

**Review by jaumeortola — APPROVED** [review]

**Pull request merged**